lib/AST/ASTMangler.cpp

```diff
@@ -2301,6 +2301,19 @@ void ASTMangler::appendImplFunctionType(SILFunctionType *fn,
     OpArgs.push_back('H');
   }
 
+  // Mangle if we have a sending result and we are in a recursive position.
+  //
+  // DISCUSSION: We only want sending results to change ABI if it is using in a
+  // function value passed to a parameter or generic position... but not if it
+  // is just added to a return type.
+  //
+  // E.x.:
+  //
+  //   func foo() -> sending X // No mangling
+  //   func bar(_ x: () -> sending X) {} // Add to mangling for x
```

Comment:
This comment implies that impl-function-type mangling is ABI, and it should probably be removed

Reply:
Yes. Sorry! I forgot to remove it.